src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -145,7 +145,7 @@ public Xml.Document parseMaven(ExecutionContext ctx) {
             return null;
         }
 
-        Set<Path> allPoms = collectPoms(mavenProject, new HashSet<>());
+        Set<Path> allPoms = collectPoms(mavenProject, new LinkedHashSet<>());
```

Comment:
Note:
The MavenParser maintains the order of poms with a LinkedHashMap, but we are passing in a `HashSet`. The HashSet causes the order to be based on the hashcode so `.iterator().next()` typically returns the wrong value. Then the resource parser adds the pom as a normal `Xml.Document`, because the `maven.getSourcePath` does not match the current project.

Alternatively, we can use a `HashSet` here, and refactor `parseMaven` to find the `Xml.Document` from the list of results returned by the MavenParser.

```java
        Xml.Document maven = mavenParserBuilder
                .build()
                .parse(allPoms, baseDir, ctx)
                .iterator() // Could stream the results and find a match based on the current project
                .next();
```

But I opted for a `LinkedHashSet` for simplicity and because there may be a lot of `Xml.Documents` to scan through, so this seems more efficient.